core/src/test/java/org/apache/calcite/test/SqlFunctionsTest.java

```diff
@@ -1054,6 +1055,54 @@ private void thereAndBack(byte[] bytes) {
       // ok
     }
   }
+  @Test void testPosition() {
+    assertThat(3, is(position("c", "abcdec")));
+    assertThat(3, is(position("c", "abcdec", 2)));
+    assertThat(3, is(position("c", "abcdec", -2)));
+    assertThat(6, is(position("c", "abcdec", 4)));
+    assertThat(6, is(position("c", "abcdec",1 , 2)));
+    assertThat(3, is(position("c", "abcdec", -1, 2)));
+    assertThat(-1, is(position("f", "abcdec", 1, 1)));
+    assertThat(-1, is(position("c", "abcdec", 1, 3)));
+    try {
+      int i = position("c", "abcdec", 0, 1);
```

Comment:
based testing 🙏

Reply:
actually made me realize those tests were wrong lol, appreciate the comment